### PR TITLE
Improve the Release Guide

### PR DESCRIPTION
**Pull request by Kerollmops**
This PR improves the release guide. It no longer refers to the `latest` branch and specifies that merge conflicts require a temporary branch.

### CODE REVIEW

File: resources/meilisearch-release.md [hunk 1]
```diff
@@ -113,9 +113,8 @@ The steps are the same as for `How to create the first pre-release (or RC i.e. R
 
 ⚙️ The CIs will be triggered to:
 - [Upload binaries](https://github.com/meilisearch/meilisearch/actions/workflows/publish-binaries.yml) to the associated GitHub release.
-- [Publish the Docker images](https://github.com/meilisearch/meilisearch/actions/workflows/publish-docker-images.yml) (`latest`, `vX.Y` and `vX.Y.Z`) to DockerHub -> check the "Docker meta" steps in the CI to check the right tags are created
+- [Publish the Docker images](https://github.com/meilisearch/meilisearch/actions/workflows/publish-docker-images.yml) (`vX.Y` and `vX.Y.Z`) to DockerHub -> check the "Docker meta" steps in the CI to check the right tags are created
 - [Publish binaries for Homebrew and APT](https://github.com/meilisearch/meilisearch/actions/workflows/publish-apt-brew-pkg.yml)
-- [Move the `latest` git tag to the release commit](https://github.com/meilisearch/meilisearch/actions/workflows/latest-git-tag.yml).
```
Comment (curquiza) [inline resources/meilisearch-release.md:118]: Why do you remove this line? The CI still exists 😇

File: resources/meilisearch-release.md [hunk 2]
```diff
@@ -113,9 +113,8 @@ The steps are the same as for `How to create the first pre-release (or RC i.e. R
 
 ⚙️ The CIs will be triggered to:
 - [Upload binaries](https://github.com/meilisearch/meilisearch/actions/workflows/publish-binaries.yml) to the associated GitHub release.
-- [Publish the Docker images](https://github.com/meilisearch/meilisearch/actions/workflows/publish-docker-images.yml) (`latest`, `vX.Y` and `vX.Y.Z`) to DockerHub -> check the "Docker meta" steps in the CI to check the right tags are created
+- [Publish the Docker images](https://github.com/meilisearch/meilisearch/actions/workflows/publish-docker-images.yml) (`vX.Y` and `vX.Y.Z`) to DockerHub -> check the "Docker meta" steps in the CI to check the right tags are created
```
Comment (curquiza) [inline resources/meilisearch-release.md:116]: Why do you remove the `latest` docker tag? Our CI pushes it

File: resources/meilisearch-release.md [hunk 3]
```diff
@@ -141,14 +140,14 @@ Instead:
...
-1. Create a new release branch starting from the latest stable Meilisearch release (`latest` or `release-vX.Y.Z`).
+1. Create a new release branch starting from the latest stable Meilisearch release (`release-vX.Y.Z`).
 
 ```bash
 # Ensure you get all the current tags of the repository
 git fetch origin --tags --force
 
 # Create the branch
-git checkout latest
+git checkout main
```
Comment (curquiza) [inline resources/meilisearch-release.md:150]: Nope, we should definitely NOT create the branch from `main`

File: resources/meilisearch-release.md [hunk 4]
```diff
@@ -141,14 +140,14 @@ Instead:
 
 It happens some releases come with impactful bugs in production (e.g. indexation or search issues): we obviously don't wait for the next cycle to fix them and we release a patched version of Meilisearch.
 
-1. Create a new release branch starting from the latest stable Meilisearch release (`latest` or `release-vX.Y.Z`).
+1. Create a new release branch starting from the latest stable Meilisearch release (`release-vX.Y.Z`).
```
Comment (curquiza) [inline resources/meilisearch-release.md:143]: Why not `latest` since the git tag exists?

File: resources/meilisearch-release.md [hunk 5]
```diff
@@ -161,13 +160,12 @@ git push -u origin release-vX.Y.Z
...
 
 ⚠️ <ins>If doing a patch release that should NOT be the `latest` release</s>:
 
 - Do NOT check `Set as the latest release` when creating the GitHub release. If you did, quickly interrupt all CIs and delete the GitHub release!
 - Once the release is created, you don't have to care about Homebrew, APT and Docker CIs: they will not consider this new release as the latest; the CIs are already adapted for this situation.
-- However, the [CI updating the `latest` git tag](https://github.com/meilisearch/meilisearch/actions/workflows/latest-git-tag.yml) is not working for this situation currently and will attach the `latest` git tag to the just-created release, which is something we don't want! If you don't succeed in stopping the CI on time, don't worry, you just have to re-run the [old CI](https://github.com/meilisearch/meilisearch/actions/workflows/latest-git-tag.yml) corresponding to the real latest release, and the `latest` git tag will be attached back to the right commit.
```
Comment (curquiza) [inline resources/meilisearch-release.md:170]: Please don't remove this line, it explains how to act if you are not doing a release that should be the latest.

**Review by curquiza — CHANGES_REQUESTED** [review]

File: resources/meilisearch-release.md (hunk 4 above)
Comment (curquiza) [inline resources/meilisearch-release.md:143]:
```suggestion
1. Create a new release branch starting from the latest stable Meilisearch release (`latest` git tag or `release-vX.Y.Z` branch).
```

File: resources/meilisearch-release.md (hunk 2 above)
Comment (Kerollmops) [reply resources/meilisearch-release.md]: I put it back too.

**Review by curquiza — APPROVED** [review]
molto bene

**Pull request merged**